docs/src/main/asciidoc/datastore.adoc

```diff
@@ -985,6 +985,36 @@ If a method's return type is `org.springframework.data.domain.Page`, the returne
 NOTE: Methods that return `Page` execute an additional query to compute total number of pages.
 Methods that return `Slice`, on the other hand, don't execute any additional queries and therefore are much more efficient.
 
+==== Null Handling of Repository Methods
+Java `java.util.Optional` can be used to indicate the potential absence of a return value.
+
+Alternatively, query methods can return the result without a wrapper.
+In that case the absence of a query result is indicated by returning `null`.
+Repository methods returning collections are guaranteed never to return `null` but rather the corresponding empty representation.
+
+==== Nullability Annotations
```

Comment:
Is this supposed to be a subsection of the above or a new section?

Reply:
I think it's a separate section.

Reply:
My question is more on "should it be"?

Reply:
I don't think it should. It's not only about return values, but also about (non-)nullable parameters.

Reply:
Whether an exception is thrown or null is returned seems directly related to the handling of empty results. Handling null parameters can be a separate section.